### PR TITLE
Add extra parameters to `exp:channel:category_archive`

### LINKED ISSUE #1256: Extra parameters on `exp:channel:category_archive`
As I started to use `exp:channel:category_archive` for more than a simple archive, some parameters are missing, like `sticky` and order by most used categories.

- [x] I am capable and would like to work on implementation of this feature if it is considered.

### PR DESCRIPTION
**Pull request by robsonsobral**
## Overview

Add `sticky` parameter and `most_used_categories` value to `orderby` parameter to `exp:channel:category_archive`.

Resolves [#1256](https://github.com/ExpressionEngine/ExpressionEngine/issues/1256).

## Nature of This Change

- [ ] 🐛 Fixes a bug
- [x] 🚀 Implements a new feature
- [ ] 🛁 Refactors existing code
- [ ] 💅 Fixes coding style
- [ ] ✅ Adds tests
- [ ] 👽 Adds new dependency
- [ ] 🔥 Removes unused files / code
- [ ] 🔒 Improves security 

## Is this backwards compatible?

- [x] Yes
- [ ] No

## Documentation

User Guide Pull Request: https://github.com/ExpressionEngine/ExpressionEngine-User-Guide/pull/231

### CODE REVIEW

**Review by intoeetive — CHANGES_REQUESTED** [review]
I think it would be good to add support for other options, that we have with channel entries, for consistency
`sticky="yes"`
`sticky="no"`
`sticky="only"`
https://docs.expressionengine.com/latest/channels/entries.html#sticky

**Comment by robsonsobral** [comment]
@intoeetive , being `no` the default value of `sticky`, what else do I need to add?! `yes` and `only` are already there. 😕

**Comment by intoeetive** [comment]
You are absolutely right. Not sure how I missed that =(

I'll add some tests to it and then we'll bake it into some future release (likely 6.2.0)

Thank you very much!

**Comment by intoeetive** [comment]
@robsonsobral I have added tests around `orderby="most_used_categories"` (did not add any for sticky parameter yet) and it looks like those are failing.
Can you have a look and fix? Or just remove this feature if this PR is mostly about `sticky` parameter (which still needs tests)

**Comment by robsonsobral** [comment]
I've no idea which drug I used when I coded that. That query makes no sense. I gonna remove `most_used_categories`.

File: tests/cypress/support/templates/cats.group/archive-sorted.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<div class="default-linear">
+    {exp:channel:category_archive channel="about" style="linear" orderby="title"}
```
Comment (TomJaeger) [inline tests/cypress/support/templates/cats.group/archive-sorted.html]: Other then the div class name above.. I don't see a difference between this and whats on line 17... Should this be running a different test? @intoeetive

File: tests/cypress/support/templates/cats.group/archive-sorted.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+<div class="default-nested">
+    {exp:channel:category_archive channel="about" style="nested" orderby="title"}
+        {categories}
+            <div>{category_name}</div>
+        {/categories}
+    {/exp:channel:category_archive}
+</div>
+
+<div class="title-linear">
+    {exp:channel:category_archive channel="about" style="linear" orderby="title"}
```
Comment (TomJaeger) [inline tests/cypress/support/templates/cats.group/archive-sorted.html]: Should this be different then whats on line 2?

**Review by TomJaeger — COMMENTED** [review]
Left 2 comments for @intoeetive above. I also noticed that you made mention of adding tests around sticky above @intoeetive I think it would be good to get them in before we merge.

**Pull request merged**